common/djangoapps/third_party_auth/saml.py

```diff
@@ -33,6 +33,17 @@ def setting(self, name, default=None):
         except KeyError:
             return self.strategy.setting(name, default)
 
+    def auth_url(self):
+        """
+        Check that the request includes an 'idp' parameter before getting the
+        URL to which we must redirect in order to authenticate the user.
+
+        raise AuthMissingParameter if the 'idp' parameter is missing.
```

Comment:
Also say "This function can be removed when the fix is merged upstream in the python-social-auth SAMLAuth's auth_url method."